cf-execd/cf-execd.c

```diff
@@ -946,13 +948,51 @@ static bool ScheduleRun(EvalContext *ctx, Policy **policy, GenericAgentConfig *c
 
         GenericAgentConfigSetBundleSequence(config, NULL);
 
+#ifndef __MINGW32__
+        /* Take over the runagent_socket_allow_users set for comparison. */
+        StringSet *old_runagent_allow_users = NULL;
+        if ((RUNAGENT_SOCKET_DIR == NULL) || (!StringEqual_IgnoreCase(RUNAGENT_SOCKET_DIR, "no")))
+        {
+            old_runagent_allow_users = (*execd_config)->runagent_allow_users;
+            (*execd_config)->runagent_allow_users = NULL;
+        }
+#endif
+
         *policy = LoadPolicy(ctx, config);
         ExecConfigDestroy(*exec_config);
         ExecdConfigDestroy(*execd_config);
 
         *exec_config = ExecConfigNew(!ONCE, ctx, *policy);
         *execd_config = ExecdConfigNew(ctx, *policy);
 
+#ifndef __MINGW32__
+        if ((RUNAGENT_SOCKET_DIR == NULL) || (!StringEqual_IgnoreCase(RUNAGENT_SOCKET_DIR, "no")))
+        {
+            if (!StringSetIsEqual(old_runagent_allow_users,
+                                  (*execd_config)->runagent_allow_users))
+            {
```

Comment:
```
// List of users in policy has changed
```

(At first I was wondering if old users referred to the old users in ACL on file).